### DISCUSSION TITLE
モデルの読み込みの機能実装についての質問

### DISCUSSION THREAD

**Discussion by nira-yusan** [pixiv/vroid-sdk-developers#134] [Q&A]
質問がいくつかあります。

質問1
ゲーム内にて、モデルIDを利用して一度読み込んだことのあるVRoidを選択せずに自動で読み込むという機能を考えています。そのため、モデルIDのセーブデータへの書き込みを考えていますが、モデルIDの公開は大丈夫なのでしょうか？

質問2
上記の質問1の自動読み込み機能についてですが、複数体（最大3体ほど）のVRoidの読み込みも考えています。その時の利用条件の表示で、読み込むVRoidの利用条件をまとめて表示してもよろしいでしょうか？
利用条件の表示についてはしっかりとガイドラインに従います。
自分のVRoidとそうではないVRoidを一括で読み込む場合の利用条件の表示は、自分のVRoidの利用条件表示欄には利用条件ではなく「自分のモデル」といったように表示し、そうでないものはガイドラインに従って利用条件を表示するつもりです。
複数体の利用条件の一括表示というケースを自分は見たことがないので、よければ聞きたいです。

質問3
上記の質問2についてですが、モデルIDを利用して自動で読み込んだモデルが自分のモデルかどうかを判別する方法はありますでしょうか？

**Comment by nyamadan** [comment]
> 質問1
> ゲーム内にて、モデルIDを利用して一度読み込んだことのあるVRoidを選択せずに自動で読み込むという機能を考えています。そのため、モデルIDのセーブデータへの書き込みを考えていますが、モデルIDの公開は大丈夫なのでしょうか？

[character_model_id](https://developer.vroid.com/sdk/docs/0.4.0/Pixiv.VroidSdk.Api.DataModel.DownloadLicense.html#Pixiv_VroidSdk_Api_DataModel_DownloadLicense_character_model_id)は保存、公開いただいて問題ございません。

> 質問2
> 上記の質問1の自動読み込み機能についてですが、複数体（最大3体ほど）のVRoidの読み込みも考えています。その時の利用条件の表示で、読み込むVRoidの利用条件をまとめて表示してもよろしいでしょうか？
利用条件の表示についてはしっかりとガイドラインに従います。
> 自分のVRoidとそうではないVRoidを一括で読み込む場合の利用条件の表示は、自分のVRoidの利用条件表示欄には利用条件ではなく「自分のモデル」といったように表示し、そうでないものはガイドラインに従って利用条件を表示するつもりです。
> 複数体の利用条件の一括表示というケースを自分は見たことがないので、よければ聞きたいです。

複数体同時に利用条件表示しているケースに心当たりはございませんが、どのモデルがどの利用条件となっているかわかるようになっていれば問題ありません。

> 質問3
> 上記の質問2についてですが、モデルIDを利用して自動で読み込んだモデルが自分のモデルかどうかを判別する方法はありますでしょうか？

通常のSDKであれば自分の読み込んだモデルであることが明らかであると思いますので、マルチプレイSDKでしょうか？
マルチプレイSDKにおいてはcharacter_model_id自分が読み込んだモデルかどうかを判別する実装は存在しません。
自分の読み込んだcharacter_model_idをストレージに保存するなど実装していただければ自分のモデルかどうかを判定できるかと思います。
